lib/postgres_to_redshift.rb

```diff
@@ -113,32 +113,67 @@ def bucket
     @bucket ||= s3.buckets[ENV['S3_DATABASE_EXPORT_BUCKET']]
   end
 
-  def copy_table(table)
+  def copy_table(table, batch_size: nil)
     tmpfile = Tempfile.new("psql2rs")
     tmpfile.binmode
     zip = Zlib::GzipWriter.new(tmpfile)
     chunksize = 5 * GIGABYTE # uncompressed
     chunk = 1
+    # Delete previous exports before starting
     bucket.objects.with_prefix("export/#{table.target_table_name}.psv.gz").delete_all
+
+    # 1. Determine the list of COPY commands to run
+    if batch_size.nil?
+      # Normal execution: Single command for the whole table
+      copy_commands = [
+        "COPY (SELECT #{table.columns_for_copy} FROM #{source_schema}.#{table.name}) TO STDOUT WITH DELIMITER ',' QUOTE '''' ENCODING 'UTF8' CSV"
+      ]
+    else
+      # Batched execution: Determine ID ranges
+      id_column = table.id_column # Assumes this method exists on 'table'
+      puts "Determining min/max ID for batched copy on column: #{id_column}"
+
+      # Get min/max ID from the source table
+      id_range_query = "SELECT MIN(#{id_column}), MAX(#{id_column}) FROM #{source_schema}.#{table.name}"
+      range_result = source_connection.exec(id_range_query).first
+      min_id = range_result['min']&.to_i
+      max_id = range_result['max']&.to_i
+
+      # If the table is empty or has no IDs, stop
+      return if min_id.nil? || max_id.nil?
```

Comment:
maybe we could raise exception here to help dev know what broke.

```suggestion
      raise "DeveloperError: Table #{table} does not have an id column." if min_id.nil? || max_id.nil?
```